src/server/replica.h

```diff
@@ -58,6 +58,7 @@ class Replica {
       std::atomic_uint32_t counter;
       TxExecutionSync(uint32_t counter) : barrier(counter), counter(counter) {
       }
+      std::vector<journal::ParsedEntry> entries_vec;
```

Comment:
This won't work because the entry doesn't own the memory, its lifetime is bound to the reader buffer (see the reader)

We can simplify it... I just thought of an allocation less execution for simple entries

Reply:
Now that the rdb loader has a pool, we can also use it here in the future

Reply:
Simplest change to make it work: https://github.com/dragonflydb/dragonfly/pull/600

Reply:
I dont understand why this will no work.
In this flow all the fibers are waiting on the barrier until the command execution is done so the reader is not continuing to read more entries.
The fibers should hold till the execution is done for the correctness of the data in db

Reply:
Oh, I see, I forgot we have a barrier after execution as well